src/dos/dos_files.cpp

```diff
@@ -804,21 +804,40 @@ bool DOS_SetFileAttr(const char* const name, FatAttributeFlags attr)
 bool DOS_Canonicalize(const char* const name, char* const big)
 {
 	// TODO Add Better support for devices and shit but will it be needed i
-	// doubt it :) 
-	uint8_t drive;
-	char fullname[DOS_PATHLENGTH];
-	if (!DOS_MakeName(name,fullname,&drive)) return false;
-	big[0]=drive+'A';
-	big[1]=':';
-	big[2]='\\';
-	strcpy(&big[3],fullname);
+	// doubt it :)
+
+	// Initalize to invalid drive index so assert will fire if DOS_MakeName
+	// does not set this
+	uint8_t drive                 = 255;
+	char fullname[DOS_PATHLENGTH] = {};
+	if (!DOS_MakeName(name, fullname, &drive)) {
+		return false;
+	}
+	big[0] = drive_letter(drive);
```

Comment:
Could you rename `big` to something meaningful?

Reply:
Renamed